### PR TITLE
Fix memory safety issue in slistartup.cc

### PR DESCRIPTION
**Pull request by heshpdx**
Also fixes the memory leak in parser.h

As referred to in https://github.com/nest/nest-simulator/issues/3217#issuecomment-2269949314

### CODE REVIEW

File: sli/parser.h [hunk 1]
```diff
@@ -61,6 +61,7 @@ class Parser
 public:
   Parser();
   Parser( std::istream& );
+  ~Parser() { if (s) delete s; }
```
Comment (heplesser) [inline sli/parser.h:64]: According to *The C++ Programming Language*, 4th ed, §11.2, "[a]pplying `delete` to the `nullptr` has no effect." So the `if` is not needed here:
```suggestion
  ~Parser() { delete s; }
```

In general, we require braces around single-line blocks (as a consequence of the OpenSSH goto-goto bug some years back).

File: sli/slistartup.cc [hunk 2]
```diff
@@ -214,53 +214,55 @@ SLIStartup::SLIStartup( int argc, char** argv )
     StringDatum* sd = new StringDatum( argv[ i ] );
     args_array.push_back( Token( sd ) );
 
-    if ( *sd == "-d" or *sd == "--debug" )
+    std::string myarg = argv[i];
```
Comment (heplesser) [inline sli/slistartup.cc:217]: Our general policy is to declare `const` for anything that does not change for clarity. We also require whitespace inside parentheses and brackets:
```suggestion
    const std::string myarg = argv[ i ];
```

**Review by heplesser — CHANGES_REQUESTED** [review]
Hi @heshpdx! Thank you for your pull request. I have added a few suggestions to get in line with NEST coding standards and solve the clang-format failure.

File: sli/parser.h [hunk 3]
```diff
@@ -61,6 +61,7 @@ class Parser
 public:
   Parser();
   Parser( std::istream& );
+  ~Parser() { delete s; }
```
Comment (heplesser) [inline sli/parser.h:64]: There is another little formatting problem which I had overlooked yesterday. This should fix it:
```suggestion
  ~Parser()
  {
    delete s;
  }
```  
BTW, we have set up pre-commit checks for NEST to make it easier for contributors to submit code in accordance to our formatting guidelines, see https://nest-simulator.readthedocs.io/en/stable/developer_space/guidelines/coding_guidelines_check.html#tooling.

Comment (heshpdx) [reply sli/parser.h:64]: Can you share why line 60 formatting is ok? I matched the destructor formatting to the constructor formatting.

Comment (heplesser) [reply sli/parser.h:64]: The destructor is defined inline, while the constructor is only declared in line 63, so there is no body, and the formatting problem related to the method body.

**Review by heplesser — APPROVED** [review]
All good now!

**Comment by heplesser** [comment]
Small fix in SLI parser plus small syntax cleanup, this merging with single review.

**Pull request merged**